pkg/kubelet/dockertools/docker.go

```diff
@@ -299,7 +299,12 @@ func ParseDockerName(name string) (dockerName *KubeletContainerName, hash uint64
 }
 
 func LogSymlink(containerLogsDir, podFullName, containerName, dockerId string) string {
-	return path.Join(containerLogsDir, fmt.Sprintf("%s_%s-%s.%s", podFullName, containerName, dockerId, LogSuffix))
+	logPath := fmt.Sprintf("%s_%s-%s.%s", podFullName, containerName, dockerId, LogSuffix)
+	// Length of a filename cannot exceed 256 characters in ext4 on Linux.
+	if len(logPath) > 255 {
+		logPath = logPath[:255]
```

Comment:
Chopping from the beginning would mean that you cannot infer what pod the logs belong to. Chopping in general is bad where ever it is done. A proper fix would be to create directory hierarchies.